### PR TITLE
Fire slack notification when candidate sends reminder to referee

### PR DESCRIPTION
**Pull request by TobyRet**
## Context
```
As a... product owner
I need to... get a feel for when and how often candidate’s are sending referee reminder emails
So that... so I can adjust the design of this feature.
```

## Changes proposed in this pull request

- Fire Slack notification when candidate sends reminder to referee
- Add specs for `RequestReference` service
- Refactor `CancelReference` service to use `SlackNotifications` module

## Guidance to review

When you send a reminder you should see a Slack request appear in the Sidekiq logs

## Link to Trello card

https://trello.com/c/dxaZlu7i/2250-%F0%9F%92%94-dev-slack-notification-when-candidate-sends-reminder-to-referee

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

**Comment by george-bit** [comment]
Refactor of CancelReferee LGTM, as does the rest of it. Will defer to senior though incase something I'm missing 👍

**Review by george-bit — COMMENTED** [review]
Refactor of CancelReferee LGTM, as does the rest of it. Will defer to senior though incase something I'm missing 👍

**Review by davidgisbey — APPROVED** [review]
I really like the new module 👍

File: spec/services/candidate_interface/decoupled_references/send_reference_reminder_spec.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    it 'updates the reference to record when the reminder was sent' do
+      expect(reference.reminder_sent_at).to eq(Time.zone.now)
+    end
+
+    it 'sends a reference chaser request to the referee mailer' do
+      expect(RefereeMailer).to have_received(:reference_request_chaser_email).with(reference.application_form, reference)
+    end
+
+    it 'sends a the slack notification worker' do
```
Comment (malcolmbaig) [inline spec/services/candidate_interface/decoupled_references/send_reference_reminder_spec.rb:32]: `sends a message to the`?

Comment (TobyRet) [reply spec/services/candidate_interface/decoupled_references/send_reference_reminder_spec.rb:32]: Yessssss

**Review by malcolmbaig — APPROVED** [review]

**Pull request merged**